File: Hard/2026-02-20-761-Special-Binary-String/solution.js
var makeLargestSpecial = function(s) {
    let count = 0, start = 0;
    const specialStrings = [];
    
    for (let i = 0; i < s.length; i++) {
        count += s[i] === '1' ? 1 : -1;
        if (count === 0) {
            specialStrings.push('1' + makeLargestSpecial(s.slice(start + 1, i)) + '0');
            start = i + 1;
        }
    }
    
    specialStrings.sort().reverse();
    return specialStrings.join('');
};

File: Hard/2026-02-20-761-Special-Binary-String/solution.py
class Solution:
    def makeLargestSpecial(self, s: str) -> str:
        count = i = 0
        special_substrings = []
        
        for j in range(len(s)):
            count += 1 if s[j] == '1' else -1
            if count == 0:
                # Recursively solve the inner special string
                special_substrings.append('1' + self.makeLargestSpecial(s[i + 1:j]) + '0')
                i = j + 1
        
        # Sort the special substrings in reverse order to get the largest lexicographical order
        special_substrings.sort(reverse=True)
        
        return ''.join(special_substrings)